plugins/artifacts/payload-schema.yml

```diff
@@ -0,0 +1,30 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: payload
+description: Artifacts to be published
+type: array
+items:
+  type: object
+  properties:
+    type:
+      title: Upload type
+      description: Artifacts can be either an individual `file` or a `directory` containing potentially multiple files with recursively included subdirectories
+      type: string
+      enum: [file, directory]
+    path:
+      title: Artifact Path
+      description: Filesystem path of artifact
+      type: string
+      pattern: "^.*[^/]$"
+    name:
+      title: Artifact Name
+      description: This will be the leading path to directories and the full name for files that are uploaded to s3
+      type: string
+      pattern: "^[^/].*[^/]$"
```

Comment:
I think that limits similar to scopes make a lot of sense.  These are basically filenames, and encodings + filenames are an absolute nightmare, so let's avoid it by sticking to printable ascii